src/bytes.ts

```diff
@@ -1,7 +1,7 @@
-import type { BytesLike } from '@ethersproject/bytes';
-type HexString = string;
+import type { BytesLike } from '@metamask/utils';
```

Comment:
The `BytesLike` type in this package is stricter, it requires strings to have the `0x` prefix at the type level. Since some of the user-facing functions of this library have `BytesLike` arguments, this is a breaking change.

We should keep the looser type and convert to `HexString` internally.

Reply:
I see the issue is that if we keep the previous definition, many functions expect a `0x` prefixed string so the code won't compile. I agree it's a breaking change but I don't think it is a big deal since ethers was expecting the same prefix anyway.

Perhaps just a changelog entry is fine. Similar to how we note breaking changes in custom errors

Reply:
Having dealt with similar errors I do see it as a big deal. Yes the strings will all have been correctly prefixed but the user has to insert a lot of type assertions or other code. It's very annoying, I would strongly avoid it until a major release.